onnxruntime/core/providers/webgpu/webgpu_context.cc

```diff
@@ -279,7 +279,11 @@ Status WebGpuContext::Run(ComputeContext& context, const ProgramBase& program) {
   uint32_t x = program.DispatchGroupSizeX();
   uint32_t y = program.DispatchGroupSizeY();
   uint32_t z = program.DispatchGroupSizeZ();
-  ORT_RETURN_IF_ERROR(program_mgr_->NormalizeDispatchGroupSize(x, y, z));
+
+  // Skip normalization for indirect dispatch since dimensions are determined by the indirect buffer
+  if (!program.UseIndirectDispatch()) {
+    ORT_RETURN_IF_ERROR(program_mgr_->NormalizeDispatchGroupSize(x, y, z));
+  }
 
   bool is_1d_dispatch = (y == 1 && z == 1);
 
```

Comment:
does "whether use indirect dispatch or not" affect the program key ?

Reply:
It depends on the op. Theoretically, using indirect dispatch or not won't impact the shader. The GQA's change is because that we want to support graph capture and meanwhile, keep the previously path unchanged. So I updated the cacheKey in the op not here. Do you prefer update here?